src/shared/components/ShareWithColleagues/ShareWithColleagues.tsx

```diff
@@ -0,0 +1,225 @@
+import {
+	Avatar,
+	Button,
+	Dropdown,
+	DropdownButton,
+	DropdownContent,
+	Icon,
+	IconName,
+	MenuContent,
+	TextInput,
+} from '@viaa/avo2-components';
+import { isEmpty, isNil, truncate } from 'lodash';
+import React, { FC, useState } from 'react';
+
+import withUser, { UserProps } from '../../hocs/withUser';
+import useTranslation from '../../hooks/useTranslation';
+
+import { shareUserRightToString, sortShareUsers } from './ShareWithColleagues.helpers';
+import './ShareWithColleagues.scss';
+import { ShareUserInfo, ShareUserInfoRights } from './ShareWithColleagues.types';
+
+type ShareWithColleaguesProps = {
+	users: ShareUserInfo[];
+	onAddNewUser: (info: Partial<ShareUserInfo>) => void;
+	onEditRights: (info: ShareUserInfo, newRights: ShareUserInfoRights) => void;
+	onDeleteUser: (info: ShareUserInfo) => void;
+};
+
+const ShareWithColleagues: FC<ShareWithColleaguesProps & UserProps> = ({
+	users,
+	user,
+	onAddNewUser,
+	onDeleteUser,
+	onEditRights,
+}) => {
+	const { tText } = useTranslation();
+	const currentUser = users.find((u) => u.email === user?.mail) as ShareUserInfo;
+	const [isRightsDropdownOpen, setIsRightsDropdownOpen] = useState<boolean>(false);
+	const [newShareUser, setNewShareUser] = useState<Partial<ShareUserInfo>>({
+		email: undefined,
+		rights: undefined,
+	});
+	const [error, setError] = useState<string | null>(null);
+
+	const handleRightsButtonClicked = () => {
+		setIsRightsDropdownOpen(!isRightsDropdownOpen);
+	};
+
+	const handleAddNewUser = () => {
+		if (/^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]+$/.test(newShareUser.email as string)) {
```

Comment:
if email is undefined this code will crash
it's better to just default to empty string, then the regex test fill execute normally and return false

We might want to extract this as a helper in the shared folder since it is also used here. Something like: isValidEmail(): boolean
```
		if (!email) {
			errors.push(`Email ${requiredError}`);
		} else if (!/^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]+$/.test(email)) {
			errors.push(
				tText(
					'authentication/views/registration-flow/r-4-manual-registration___email-is-geen-geldig-email-adres'
				)
			);
		}
```

```suggestion
		if (/^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]+$/.test(newShareUser.email || '')) {
```